### PR TITLE
Student entries

### PR DESCRIPTION
**Pull request by davidabrahams**

### CODE REVIEW

**Comment by davidabrahams** [comment]
Merge conflicts are resolved! We should all look over this really carefully since there's a ton of new code to integrate. I tried really hard to make sure not to delete someone's functionality, but I can't promise I didn't make a mistake.

File: routes/index.js [hunk 1]
```diff
@@ -18,79 +24,186 @@ routes.POSTlogin = function(req, res, next) {
...
+  // var currentDate = new Date();
+  var studentID = req.params._id;
+  var studentAttendance = req.body.attendance;
+  var studentEntry = req.body.entry;
+  var studentGrades = req.body.grades;
+  var date = req.body.date;
+  console.log("formdata", studentID, studentAttendance, studentGrades,
+    studentEntry, date)
+
+  Attendance.create({
```
Comment (nmohamed) [inline routes/index.js:152]: http://wildermuth.com/2013/8/3/JavaScript_Promises
for async calls, i don't remember it well, but might be useful

File: public/javascripts/ng-apps.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    })
+    .when('/login', {
+      templateUrl: 'views/login.html',
+      controller: 'loginCtrl'
+    })
+    .when('/', {
+      templateUrl: 'views/login.html',
+      controller: 'loginCtrl'
+    })
+});
```
Comment (shrutiyer) [inline public/javascripts/ng-apps.js:17]: WIll this file contain only the routes and app declaration?

File: public/views/login.html [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      </div>
+      <div class="form-group" id="password-group">
+        <label for="inputPassword" class="sr-only">Password</label>
+        <input type="password" id="inputPassword" class="form-control" placeholder="Password" ng-model="password" required>
+        <span ng-show="password_error" class="help-block">{{password_error}}</span>
+      </div>
+      <div class="checkbox">
+        <label>
+        <input type="checkbox" value="remember-me"> Remember me
+        </label>
```
Comment (shrutiyer) [inline public/views/login.html:18]: Not really a comment on this PR:
Making the Remember me feature requires sending cookies to the user. Reading about it, I realize it should be a stretch goal.

File: routes/index.js [hunk 4]
```diff
@@ -18,79 +24,186 @@ routes.POSTlogin = function(req, res, next) {
...
+
+      // send email verification
+      // Used example: https://github.com/heitortsergent/passport-local-mongoose-email/tree/master/examples/login
+      // req.headers.host gets the current host url
+      var authenticationURL = 'http://' + req.headers.host +
+        '/verify?authToken=' + user.authToken;
+      var email = new sendgrid.Email();
+
+      email.addTo(user.email);
+      email.setFrom('SpringInitiative@olinjs.com');
```
Comment (shrutiyer) [inline routes/index.js:54]: Should we make a gmail account to be the admin? They can even authorize people and other admin-duties

**Comment by shrutiyer** [comment]
Alright, I read through the PR. I think I understand the structure. I can't run it but I trust you guys!
Thumbs up from me

**Pull request merged**